### PR TITLE
Initial version.

### PR DESCRIPTION
**Pull request by dartartem**

### CODE REVIEW

File: docker-compose-mysql-binlog.yml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    image: eventuateio/eventuate-mysql:$EVENTUATE_COMMON_VERSION
+    ports:
+      - 3306:3306
+    environment:
+      - MYSQL_ROOT_PASSWORD=rootpassword
+      - MYSQL_USER=mysqluser
+      - MYSQL_PASSWORD=mysqlpw
+
+  cdcservice:
+    image: test-eventuate-cdc-service
```
Comment (cer) [inline docker-compose-mysql-binlog.yml]: Please use correct image!

File: docker-compose-mysql-binlog.yml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+      EVENTUATELOCAL_CDC_DB_PASSWORD: rootpassword
+      EVENTUATELOCAL_CDC_READER_NAME: MySqlReader
+      EVENTUATELOCAL_CDC_OFFSET_STORE_KEY: MySqlBinlog
+      EVENTUATELOCAL_CDC_MYSQL_BINLOG_CLIENT_UNIQUE_ID: 1234567890
+      EVENTUATE_CDC_KAFKA_ENABLE_BATCH_PROCESSING: "true"
+      EVENTUATELOCAL_CDC_READ_OLD_DEBEZIUM_DB_OFFSET_STORAGE_TOPIC: "false"
+
+
+  adminer:
+    image: adminer
```
Comment (cer) [inline docker-compose-mysql-binlog.yml]: Please specify a version

File: gradle.properties [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+
+eventuateMavenRepoUrl=https://dl.bintray.com/eventuateio-oss/eventuate-maven-release/,https://dl.bintray.com/eventuateio-oss/eventuate-maven-milestone,file:///Users/cer/.m2/testdeploy,https://dl.bintray.com/eventuateio-oss/eventuate-maven-rc
+
+eventuateTramVersion=0.23.0.MICRONAUT.M8
+dockerComposePluginVersion=0.4.5
+version=0.1.0-SNAPSHOT
+
+eventuateCommonImageVersion=0.8.0.MICRONAUT.M6
+eventuateMessagingKafkaImageVersion=0.3.0.MICRONAUT.M6
+eventuateCdcImageVersion=0.6.0.RC2
```
Comment (cer) [inline gradle.properties]: Should be RC3

**Review by cer — CHANGES_REQUESTED** [review]
Please add: 

* Tests
* CircleCI configuration

File: .circleci/config.yml [hunk 4]
```diff
@@ -0,0 +1,36 @@
+version: 2
+jobs:
+  build:
+    machine: true
+    working_directory: ~/eventuate-tram-examples-vanilla-event-publishing
+    steps:
+      - checkout
+      - restore_cache:
```
Comment (cer) [inline .circleci/config.yml:8]: See this version of restore cache, which has multiple keys: https://github.com/microservices-patterns/ftgo-application/blob/9f85c7744bfa2d58a2a9aefbb3b9eb43428df3ec/.circleci/config.yml#L8

This should be used in config.yml

File: message-publisher/src/main/java/io/eventuate/examples/tram/vanilla/event/publishing/JdbcStatementExecutor.java [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.util.List;
+import java.util.Map;
+
+public class JdbcStatementExecutor implements EventuateJdbcStatementExecutor {
+
+  private HikariDataSource dataSource;
+
+  public JdbcStatementExecutor(String user, String password, String driver, String url) {
```
Comment (cer) [inline message-publisher/src/main/java/io/eventuate/examples/tram/vanilla/event/publishing/JdbcStatementExecutor.java:19]: I think that `JdbcStatementExecutor` should be injected with a

```
interface JdbcConnectionSupplier {
  void doWithConnection(Consumer<Connection> connectionConsumer);
}
```

All the code such as `new HikariDataSource();` and `try(Connection connection = dataSource.getConnection();` should be moved into a concrete implementation of `JdbcConnectionSupplier`.

In a legacy application `JdbcStatementExecutor` would need to use the connection provided by the legacy application.

File: .circleci/config.yml (hunk 4 above)
Comment (dartartem) [reply .circleci/config.yml:8]: ok

File: message-publisher/src/main/java/io/eventuate/examples/tram/vanilla/event/publishing/JdbcStatementExecutor.java (hunk 5 above)
Comment (dartartem) [reply message-publisher/src/main/java/io/eventuate/examples/tram/vanilla/event/publishing/JdbcStatementExecutor.java:19]: ok

**Pull request merged**